analyses/snv-callers/scripts/03-calculate_tmb.R

```diff
@@ -143,46 +141,109 @@ if (!dir.exists(opt$output)) {
+
+if (opt$tcga) {
+  strelka_mutect_mnv <- strelka_mutect_mnv %>%
+    dplyr::mutate(Tumor_Sample_Barcode = substr(Tumor_Sample_Barcode, 0, 12))
+}
 
+# Split up MNVs
+strelka_mutect_maf_df <- strelka_mutect_maf_df %>%
+  dplyr::union_all(strelka_mutect_mnv,
+                   copy = TRUE
+  ) %>%
+  as.data.frame()
+
+########################### Set up metadata columns ############################
 # Print progress message
 message("Setting up metadata...")
 
-# Isolate metadata to only the samples that are in the datasets
-metadata <- readr::read_tsv(opt$metadata) %>%
-  dplyr::filter(Kids_First_Biospecimen_ID %in% maf_df$Tumor_Sample_Barcode) %>%
-  dplyr::distinct(Kids_First_Biospecimen_ID, .keep_all = TRUE) %>%
-  dplyr::rename(Tumor_Sample_Barcode = Kids_First_Biospecimen_ID)
+# Have to handle TCGA and PBTA metadata differently
+if (opt$tcga) {
+  # Format two fields of metadata for use with functions
+  metadata <- readr::read_tsv(opt$metadata) %>%
+    dplyr::mutate(
+      experimental_strategy = "WXS", # This field doesn't exist for this data, but all is WXS
+      short_histology = Primary_diagnosis
+    ) # This field is named differently
 
-# Make sure that we have metadata for all these samples.
-if (!all(unique(maf_df$Tumor_Sample_Barcode) %in% metadata$Tumor_Sample_Barcode)) {
-  stop("There are samples in this MAF file that are not in the metadata.")
+  # Manifest files only have first 12 letters of the barcode so we gotta chop the end off
+  strelka_mutect_maf_df <- strelka_mutect_maf_df %>%
+    dplyr::mutate(Tumor_Sample_Barcode = substr(Tumor_Sample_Barcode, 0, 12))
+} else {
+  # Isolate metadata to only the samples that are in the datasets
+  metadata <- readr::read_tsv(opt$metadata) %>%
+    dplyr::filter(Kids_First_Biospecimen_ID %in% strelka_mutect_maf_df$Tumor_Sample_Barcode) %>%
+    dplyr::distinct(Kids_First_Biospecimen_ID, .keep_all = TRUE) %>%
+    dplyr::rename(Tumor_Sample_Barcode = Kids_First_Biospecimen_ID)
+
+  # Make sure that we have metadata for all these samples.
+  if (!all(unique(strelka_mutect_maf_df$Tumor_Sample_Barcode) %in% metadata$Tumor_Sample_Barcode)) {
+    stop("There are samples in this MAF file that are not in the metadata.")
+  }
 }
-
-# Add the experimental strategy column on the data.frame for calculating purposes
-maf_df <- maf_df %>%
+# Add in metadata
+strelka_mutect_maf_df <- strelka_mutect_maf_df %>%
+  as.data.frame() %>%
   dplyr::inner_join(metadata %>%
-    dplyr::select(
-      Tumor_Sample_Barcode,
-      experimental_strategy,
-      short_histology
-    ))
+                      dplyr::select(
+                        "Tumor_Sample_Barcode",
+                        "experimental_strategy",
+                        "short_histology"
+                      ))
 
-############################# Calculate TMB ####################################
+# .x is messing up the maf_to_granges function
+colnames(strelka_mutect_maf_df) <- gsub("\\.x$", "", colnames(strelka_mutect_maf_df))
```

Comment:
This should only be added if there are repeated column names that are not used for the join, right? I'm confused why they are showing up at all.

Reply:
IDK, This is something we had previously too: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/b1b73fe321a97fa82d85c86d20bd85635aabba25/analyses/snv-callers/scripts/03-calculate_tmb.R#L278 It shows up after the union_all step.

Reply:
I think we want union instead of union_all http://www.datasciencemadesimple.com/union-union_all-function-r-using-dplyr-union-data-frames/

Reply:
I think I've made the changes we need. I switched to `union` and added in some `selects`  See if what I did seems reasonable/more right to you.

Reply:
I don't think it should matter if it is `union` or `union_all`, as there should not be any shared sites. Pretty sure I used union_all for speed, but I'm sure it doesn't matter much. The by= is there so we keep all of the columns that might not have equal values, one of which is the send position, which is probably what is screwing up the maf_to_granges.